common/api/core-react.api.md

```diff
@@ -122,6 +121,13 @@ export interface BadgeProps extends CommonProps {
     svg: any;
 }
 
+// @public
+export enum BadgeType {
```

Comment:
Ideally I'd inline this in the component props (so we can i.e. add new union values).
However, since `BadgeType` is used in multiple APIs it will get quite repetitive.
Either way, let's avoid adding enums and use string literal types instead.
Ideally with the capability to add support for additional/custom badge types in the future (in non breaking way w/o introducing additional types).
I.e. does this make sense? `type BadgeType = "new" | "technical-preview" | (string & {});`
https://www.typescriptlang.org/play?#code/PTAEDkFNIE10BmB7ATqAlgOwC6QDZ7oDOkmJAUNgJ4AOkoAgplQMrYpYDmoAvKABRF2XUADJQAbwC+ASgDc5SAA8aqbKGp1QAIQCGMTpAAqtenwBEmSAHdzoAD6hzuAMYALTOhe68AWhookABu6DZ2jkyswpicCuTkCACumC7Y6EiYGpBC-EoAXDr6hiZ0MpLkUvG4OZZh8pTZ2PyWSNi+nIG6bQBGiW3WkL7emL76ML5W1r7dRdmjRL4DBOYy8UA

Reply:
Today I learned that is the recommended way to not widen string literals inside of a union with other strings. Hoped there would be an `as const` or something. Agree that sounds good.

Reply:
done - interesting trick